### PR TITLE
WIP global: refactoring for Invenio 3

### PR DESCRIPTION
**Pull request by SamiHiltunen**
https://github.com/zenodo/invenio/tree/zenodo-master/invenio/modules/pages

Co-authored-by: Lars Holm Nielsen <lars.holm.nielsen@cern.ch>
Signed-off-by: Sami Hiltunen <sami.mikael.hiltunen@cern.ch>

### CODE REVIEW

File: requirements-devel.txt [hunk 1]
```diff
@@ -21,8 +21,9 @@
 # In applying this license, CERN does not
 # waive the privileges and immunities granted to it by virtue of its status
 # as an Intergovernmental Organization or submit itself to any jurisdiction.
-
--e git+git://github.com/inveniosoftware/invenio-base.git#egg=invenio-base
--e git+git://github.com/inveniosoftware/invenio-ext.git#egg=invenio-ext
--e git+git://github.com/inveniosoftware/invenio-testing.git#egg=invenio-testing
--e git+git://github.com/inveniosoftware/invenio-upgrader.git#egg=invenio-upgrader
+#
```
Comment (jirikuncar) [inline requirements-devel.txt]: invenio-db

File: setup.py [hunk 2]
```diff
@@ -22,38 +22,52 @@
...
+for reqs in extras_require.values():
+    extras_require['all'].extend(reqs)
+
+setup_requires = [
+    'Babel>=1.3',
 ]
 
+install_requires = [
+    'Flask-BabelEx>=0.9.2',
+    'Invenio-DB>=1.0.0a7'  # TODO: Bump version when versioning support is out
```
Comment (jirikuncar) [inline setup.py]: `s/Invenio-DB/invenio-db/`

File: tests/test_models.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+from __future__ import absolute_import, print_function
+
+from invenio_pages import InvenioPages, Page
+from invenio_pages.views import blueprint
+
+
+def test_page_repr(pages_fixture):
+    app = pages_fixture
+    InvenioPages(app)
```
Comment (jirikuncar) [inline tests/test_models.py]: this should not be necessary - https://github.com/inveniosoftware/invenio-pages/pull/15/files#diff-484462fced51d1a06b1d93b4a44dd535R51

Comment (SamiHiltunen) [inline tests/test_models.py]: Some of the tests need to control the initialization, so I ended up removing the initialization from the fixture and allow the tests to initialize the extension by themselves.

**Comment by jirikuncar** [comment]
Superseded by #16.

**Pull request closed without merging**